xmind/tests/test_topic_element.py

```diff
@@ -469,3 +469,377 @@ def test_getMarkers(self):
         _marker_fefs_element.getChildNodesByTagName.assert_called_once_with(
             TAG_MARKERREF)
         self._assert_init_methods()
+
+    def test_addMarker_markerId_is_none(self):
+        _element = TopicElement()
+
+        _get_markerrefs = patch.object(_element, '_get_markerrefs').start()
+
+        self.assertIsNone(_element.addMarker(None))
+        _get_markerrefs.assert_not_called()
+        self._assert_init_methods()
+
+    def test_addMarker_markerId_is_str(self):
+        _element = TopicElement()
+
+        _get_markerrefs = patch.object(_element, '_get_markerrefs').start()
+        _get_markerrefs.side_effect = Exception
+
+        _marker_refs_element_constructor_mock = self._init_patch_with_name(
+            '_marker_refs_element_constructor_mock',
+            'xmind.core.topic.MarkerRefsElement'
+        )
+
+        _marker_id_constructor = self._init_patch_with_name(
+            '_marker_id_constructor',
+            'xmind.core.topic.MarkerId',
+            return_value='new_marker_id'
+        )
+
+        with self.assertRaises(Exception) as _ex_mock:
+            _element.addMarker('marker_test')
+
+        _get_markerrefs.assert_called_once()
+        _marker_refs_element_constructor_mock.assert_not_called()
+        _marker_id_constructor.assert_called_once_with('marker_test')
+        self._assert_init_methods()
+
+    def test_addMarker_markerId_is_object(self):
+        _element = TopicElement()
+
+        _get_markerrefs = patch.object(_element, '_get_markerrefs').start()
+        _get_markerrefs.side_effect = Exception
+
+        _marker_refs_element_constructor_mock = self._init_patch_with_name(
+            '_marker_refs_element_constructor_mock',
+            'xmind.core.topic.MarkerRefsElement'
+        )
+
+        _marker_id_constructor = self._init_patch_with_name(
+            '_marker_id_constructor',
+            'xmind.core.topic.MarkerId',
+            return_value='new_marker_id'
+        )
+
+        with self.assertRaises(Exception) as _ex_mock:
```

Comment:
same as above. Valid for all unused _ex_mock(s)

Reply:
Addressed